tests/nativemessaging/helper.cpp

```diff
@@ -6,29 +6,38 @@
 
 QVector<QObject*> TestHelper::s_testList;
 QProcess* TestHelper::s_nativeMessagingProcess = nullptr;
+char* TestHelper::s_app = nullptr; 
+int TestHelper::s_last_exit_code = 0; 
 
 TestHelper::TestHelper() { s_testList.append(this); }
 
 // static
-void TestHelper::runNativeMessaging(const char* app) {
+void TestHelper::runNativeMessaging(const char* app, QStringList arguments) {
   Q_ASSERT(s_nativeMessagingProcess == nullptr);
 
   s_nativeMessagingProcess = new QProcess();
   s_nativeMessagingProcess->setReadChannel(QProcess::StandardOutput);
+  s_last_exit_code=0;
 
   connect(s_nativeMessagingProcess, &QProcess::readyReadStandardError, []() {
     qDebug() << "[mozillavpnnp - stderr]"
              << s_nativeMessagingProcess->readAllStandardError();
   });
-  s_nativeMessagingProcess->start(app, QStringList(),
+  connect(s_nativeMessagingProcess, &QProcess::finished, [&](int exitCode) {
+    s_last_exit_code=exitCode;
+  });
+  s_nativeMessagingProcess->start(app, arguments,
                                   QProcess::Unbuffered | QProcess::ReadWrite);
   if (!s_nativeMessagingProcess->waitForStarted()) {
-    qFatal("Failed to start the naive messaging process");
+    // qFatal("Failed to start the naive messaging process");
```

Comment:
Why comment this out?

Reply:
Oops, was debugging exceptions. Removed that :)